### ISSUE TITLE
error configuring CNI network plugin: exec: "nsenter": executable file not found in $PATH

### ISSUE THREAD

**Issue by statiksof** [containers/podman#12152]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

1. podman run -d -it centos:7

2. podman exec -it container ping www.google.com

**Describe the results you received:**

```
Error: error configuring CNI network plugin: exec: "nsenter": executable file not found in $PATH
``` 

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
3.2.3
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.2.3-0.11.module_el8.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Comment by flouthoc** [comment]
I think its missing following package `util-linux` could you please try installing with `yum install util-linux`.

**Comment by statiksof** [comment]
@flouthoc where? on the host package `util-linux` is already installed.

**Comment by flouthoc** [comment]
@statiksof what is the output of `which nsenter` on host.

**Comment by flouthoc** [comment]
also does error occurs after step 1 or step 2 ?

**Comment by statiksof** [comment]
> also does error occurs after step 1 or step 2 ?

after 2

**Comment by statiksof** [comment]
> @statiksof what is the output of `which nsenter` on host.

somehow I am having problems running `which` in the container (command not found) but on the host the output is `/usr/bin/nsenter`

**Comment by flouthoc** [comment]
@statiksof on **host** does `echo $PATH` contains path `/usr/bin/` or `/usr/bin/nsenter`. If not could you please add path to PATH variable and try again.

Also we don't need `nsenter` on container only on host.

**Comment by statiksof** [comment]
@flouthoc PATH contains `/usr/bin`, I added manually `/usr/bin/nsenter` but didn't change anything.

**Comment by Luap99** [comment]
`nsenter` must be in $PATH on the host and not the container.
Can you execute `nsenter` in your shell?

**Comment by statiksof** [comment]
@Luap99 yes, I can.

**Comment by Luap99** [comment]
@statiksof What is the output of `command -v nsenter` on your host.

**Comment by statiksof** [comment]
@Luap99 I am running in rootless mode, `command -v nsenter` for the user returns `/bin/nsenter`.

**Comment by statiksof** [comment]
I also activated debug mode and there are other errors/warnings before `nsenter`:

```
level=debug msg="Not configuring container store" time="2021-11-01T16:08:59+01:00" level=debug msg="Initializing event backend file" time="2021-11-01T16:08:59+01:00" level=debug msg="configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument" time="2021-11-01T16:08:59+01:00" level=debug msg="configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument" time="2021-11-01T16:08:59+01:00" level=debug msg="configured OCI runtime crun initialization failed: no valid executable found for OCI runtime crun: invalid argument" time="2021-11-01T16:08:59+01:00" level=debug msg="Using OCI runtime \"/usr/bin/runc\"" Error: error configuring CNI network plugin: exec: "nsenter": executable file not found in $PATH) 
```

**Comment by Luap99** [comment]
You can ignore the other warnings in the log.
Is `/bin` in your $PATH? How do you execute the `podman exec` command. It does not makes sense that this error only occurs for `podman exec`, the same code is used for `podman run`.

**Comment by statiksof** [comment]
`/bin` is in the $PATH. After rebooting the machine I no longer can see the container, is this related?

```
podman ps -a
ERRO[0000] error joining network namespace for container ae18061d6e35368780eb384f44932ffbfce294301ae6a304fc43132beb72335b: error retrieving network namespace at /tmp/podman-run-500/netns/cni-9d9c6762-6e01-936a-1bb3-1c0b6b3dbe88: unknown FS magic on "/tmp/podman-run-500/netns/cni-9d9c6762-6e01-936a-1bb3-1c0b6b3dbe88": 58465342 
Error: error joining network namespace of container ae18061d6e35368780eb384f44932ffbfce294301ae6a304fc43132beb72335b: error retrieving network namespace at /tmp/podman-run-500/netns/cni-9d9c6762-6e01-936a-1bb3-1c0b6b3dbe88: unknown FS magic on "/tmp/podman-run-500/netns/cni-9d9c6762-6e01-936a-1bb3-1c0b6b3dbe88": 58465342
```

**Comment by Luap99** [comment]
> `/bin` is in the $PATH. After rebooting the machine I no longer can see the container, is this related?
> 
> ```
> podman ps -a
> ERRO[0000] error joining network namespace for container ae18061d6e35368780eb384f44932ffbfce294301ae6a304fc43132beb72335b: error retrieving network namespace at /tmp/podman-run-500/netns/cni-9d9c6762-6e01-936a-1bb3-1c0b6b3dbe88: unknown FS magic on "/tmp/podman-run-500/netns/cni-9d9c6762-6e01-936a-1bb3-1c0b6b3dbe88": 58465342 
> Error: error joining network namespace of container ae18061d6e35368780eb384f44932ffbfce294301ae6a304fc43132beb72335b: error retrieving network namespace at /tmp/podman-run-500/netns/cni-9d9c6762-6e01-936a-1bb3-1c0b6b3dbe88: unknown FS magic on "/tmp/podman-run-500/netns/cni-9d9c6762-6e01-936a-1bb3-1c0b6b3dbe88": 58465342
> ```

Is your /tmp on tmpfs? The podman directories under /tmp must absolutely be removed after reboot otherwise serious bugs can happen.
You should also make sure that you have a proper logins session, e.g. do not use sudo/su.

**Comment by Luap99** [comment]
I close this issue as the main branch no longer depends on `nsenter`.
Either way something is wrong in your environment. $PATH should be the same for podman exec and podman run,  so it does not makes sense that only exec is failing.

**Comment by bdoms** [comment]
This issue should not be re-opened if the new version doesn't depend on `nsenter` but for anyone else finding this when searching for a solution like I did: I ran into this problem when trying to run `podman pod create` as a rootless user. it worked when I ran it from my own shell, but for whatever reason when it was run within a script as the exact same user I ran into this same error. Didn't have any issues with any other podman commands, except for `podman network create` which I had not previously been using.

I managed to fix it by prepending the `$PATH` environment variable like so:

```bash
PATH=$PATH podman pod create
```

I have no idea why the `$PATH` wasn't visible within this specific situation (rootless and scripted) but the fix works, so I'm using this until I can upgrade versions.
